packages/hardhat/contracts/ContributionsNFT.sol

```diff
@@ -22,9 +23,20 @@ contract ContribuData is Ownable {
     function getContributionAmounts(uint nftId, uint8 contributionType) public view returns(uint amount) {
         return contributionAmounts[nftId][contributionType];
     }
-    function getContributionTypes(uint id) public view returns(string memory name) {
+    function getContributionType(uint id) public view returns(string memory name) {
         return contributionTypes[id];
     }
+    function getContributionTypesAmount() public view returns(uint) {
+        return contributionTypesAmount;
+    }
+    function getContributionTypesIds() public view returns(uint[] memory) {
+        uint[] memory ids = new uint[](contributionTypesAmount);
+        for(uint i=0; i<contributionTypesAmount; i++)
+        {
+            ids[i] = i;
```

Comment:
It wouldn't be better return also the names?
[
{
id: number,
name: string
}
]

?

Reply:
Yes, we could potentially add a function that only returns the strings (with no Ids) to make it easier for the frontend to implement.

However, the optimal way would be to not have this function at all and use [multicall](https://docs.chainstack.com/docs/http-batch-request-vs-multicall-contract#:~:text=A%20multicall%20contract%20is%20a,parameters%20and%20executes%20them%20together.) instead.

Happy to modify this function if it helps @edsphinx